drivers/pppos.c

```diff
@@ -448,46 +477,21 @@ static void pppos_mainLoop(void* _state)
 			log_info("open success!");
 		}
 
-		serial_set_non_blocking(state->fd);
-		if (at_disconnect(state->fd) != AT_RESULT_OK)
+		if (at_setup(state)) {
+			serial_close(state->fd);
 			goto fail;
-
-		mutexLock(state->lock);
-		while (!state->apn[0])
-			condWait(state->cond, state->lock, 0);
-		mutexUnlock(state->lock);
-
-		const char** at_cmd = at_init_cmds;
-		while (*at_cmd) {
-			if ((res = at_send_cmd(state->fd, *at_cmd, 3000)) != AT_RESULT_OK) {
-				log_warn("failed to initialize modem (cmd=%s), res=%d, retrying", *at_cmd, res);
-				goto fail;
-			}
-
-			at_cmd += 1;
 		}
 
-		{ /* Configure APN */
-			char at_set_apn[256];
-			if (snprintf(at_set_apn, sizeof(at_set_apn), "AT+CGDCONT=1,\"IP\",\"%s\"\r\n", state->apn) >= sizeof(at_set_apn)) {
-				log_error("APN name too long");
-				goto fail;
-			}
-
-			if ((res = at_send_cmd(state->fd, at_set_apn, 3000)) != AT_RESULT_OK) {
-				log_warn("failed to set APN, retrying");
-				goto fail;
-			}
+		if ((res = at_send_cmd(state->fd, "AT+CGATT=0\r\n", 3000)) != AT_RESULT_OK) {
```

Comment:
I wonder if this timeout is not to small. When I read different modem documentation and chat script examples  I checked that modems requires wait from 10 to 300 seconds. If i read correctly we are waiting only 3 seconds.
What do you think @nalajcie ?

Reply:
I'm not sure why this command is needed (I don't know the exact modem version either, so I can't check it in datasheet) - maybe without it the `AT+CGDATA` fails?

This command usually is used to disconnect GPRS context, so it should be pretty quick.

If You meant the next command (`AT+CGDATA`) I think it should return `CONNECT` immediately and start connecting (next states will be available via PPP callback), but it might be up to manufacturer to mess this up :).

For LTE modems it should be pretty quick, as the primary PDP context is received and attached during network registration (NB-IoT may be similar).

I don't see any command for checking if we're even registered in the network (previously it was not enforced/waited for, just outputted to the log, as we're in an infinite loop).

Reply:
The reason for inclusion of `AT+CGATT=0` was that the modem had been attaching to a PDP context automatically and in result did not allow PPP. I have replaced `AT+CGATT=0` with `AT+QCFG="autopdn",0` and now it seems the connection is established faster. The debug commands (like `AT+CREG?`) are back.